### PR TITLE
refactor shap-ttolbar and add support for remote plugin

### PR DESCRIPTION
**Pull request by iamksuresh**
## Description

1. main.py as the main entry point
2. moved test assests to stock-plugins root (stock-plugins/user_defined_files)
3. refactored the folder and files structure.
4. Users can now execute the plugin as a module
5. Plugin will now accept comma separated assets paths (data,model,ground_truth), model classifications, ground truth value,plugin args

## Type of Change

 refactor: Code refactoring without changes in functionality

perf: Performance improvements

## How to Test

Execute the below commands to run unit tests -

```
cd aiverify/stock-plugins/aiverify.stock.shap-toolbox/algorithms/shap_toolbox/
pytest . -vv

```
image

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [x]  I have added or updated the necessary documentation (README, API docs, etc.).
- [x]  I have added appropriate unit tests or functional tests for the changes made.
- [x]  I have followed the project's coding conventions and style guidelines.
- [x]  I have rebased my branch onto the latest commit of the main branch.
- [x]  I have squashed or reorganized my commits into logical units.
- [x]  I have added any necessary dependencies or packages to the project's build configuration.
- [x]  I have performed a self-review of my own code.
- [x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

**Comment by iamksuresh** [comment]
@imda-peckyoke  / @timlrx : Tests for multiclass_non_pipeline  is failing. This will be revisited later. For now, I have commented this test.
Ref- https://github.com/aiverify-foundation/aiverify/pull/323/files#diff-0a6151e73ecee45f03ebdc7c1c53deb3ce79b44657b279d466b331da60c6e9cbR107

File: stock-plugins/aiverify.stock.shap-toolbox/algorithms/shap_toolbox/aiverify_shap_toolbox/plugin_init.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    parser.add_argument(
+        "--core_modules_path",
+        default="",
+        help="Path to the core modules (default: empty).",
+    )
+    parser.add_argument(
+        "--explain_type",
+        required=True,
+        choices=["global", "local"],
+        help="Type of Explainability.(global or local)",
```
Comment (timlrx) [inline stock-plugins/aiverify.stock.shap-toolbox/algorithms/shap_toolbox/aiverify_shap_toolbox/plugin_init.py:54]: small nit - "Type of Explainability (global or local)."

**Review by timlrx — COMMENTED** [review]
Just a very small nit on my end, otherwise looks good to me, thanks.

**Review by imda-peckyoke — APPROVED** [review]
approve

**Pull request merged**